docs/pages/config-plugins/plugins.mdx

```diff
@@ -0,0 +1,382 @@
+      if (!mainApplication['meta-data']) {
+        mainApplication['meta-data'] = [];
+      }
+
+      // Add the custom message as a meta-data entry
+      mainApplication['meta-data'].push({
+        $: {
+          'android:name': 'HelloWorldMessage',
+          'android:value': message,
+        },
+      });
+    }
+
+    return config;
+  });
+};
+
+export default withAndroidPlugin;
+```
+
+The example code above adds a meta-data entry `HelloWorldMessage` to the **android/app/src/main/AndroidManifest.xml** file by importing `ConfigPlugin` and `withAndroidManifest` from the `expo/config-plugins` library. The [`withAndroidManifest`](/config-plugins/mods/#mod-plugins) mod plugin is an asynchronous function that accepts a config and a data object and modifies the value before returning an object.
+
+</Step>
+
+<Step label="2">
+
+### Create iOS plugin
+
+In **withIosPlugin.ts**, add the following code:
+
+```ts withIosPlugin.ts
+import { ConfigPlugin, withInfoPlist } from 'expo/config-plugins';
+
+const withIosPlugin: ConfigPlugin = config => {
+  // Define the custom message
+  const message = 'Hello world, from Expo plugin!';
+
+  return withInfoPlist(config, config => {
+    // Add the custom message to the Info.plist file
+    config.modResults.HelloWorldMessage = message;
+    return config;
+  });
+};
+
+export default withIosPlugin;
+```
+
+The example code above adds `HelloWorldMessage` as the custom key with a custom message in **ios/\<your-project-name\>/Info.plist** file by importing the `ConfigPlugin` and `withInfoPlist` from the `expo/config-plugins` library. The [`withInfoPlist`](/config-plugins/mods/#mod-plugins) mod plugin is an asynchronous function that accepts a config and a data object and modifies the value before returning an object.
+
+</Step>
+
+<Step label="3">
+
+### Create a combined plugin
+
+Now you can create a combined plugin that applies both platform-specific plugins. This approach allows the maintenance of platform-specific code separately while providing a single entry point.
+
+In **withPlugin.ts**, add the following code:
+
+```ts withPlugin.ts
+import { ConfigPlugin } from 'expo/config-plugins';
+import withAndroidPlugin from './withAndroidPlugin';
+import withIosPlugin from './withIosPlugin';
+
+const withPlugin: ConfigPlugin = config => {
+  // Apply Android modifications first
+  config = withAndroidPlugin(config);
+  // Then apply iOS modifications and return
+  return withIosPlugin(config);
+};
+
+export default withPlugin;
+```
+
+</Step>
+
+<Step label="4">
+
+### Add TypeScript support and convert to dynamic app config
+
+While it is recommended that you write config plugins in TypeScript, as this will provide intellisense for the configuration objects, your app config is ultimately evaluated by Node.js, which does not recognize TypeScript code by default. You need to add a parser for the TypeScript files from the the **plugins** directory to **app.config.ts** file.
```

Comment:
```suggestion
We recommend writing config plugins in TypeScript, since this will provide intellisense for the configuration objects. However, your app config is ultimately evaluated by Node.js, which does not recognize TypeScript code by default. Therefore, you will need to add a parser for the TypeScript files from the the **plugins** directory to **app.config.ts** file.
```
Trying to remove passive voice, but not sure if I used "we" well :)